base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,871 @@
+        return speed_dict
+
+    @api.model
+    def _match_analytic_account(
+            self, aaccount_dict, chatter_msg, speed_dict=None):
+        """Example:
+        aaccount_dict = {
+            'code': '627',
+            }
+        speed_dict is usefull to gain performance when you have a lot of
+        analytic accounts to match
+        """
+        if not aaccount_dict:
+            aaccount_dict = {}
+        aaao = self.env['account.analytic.account']
+        if speed_dict is None:
+            speed_dict = self._prepare_analytic_account_speed_dict()
+        self._strip_cleanup_dict(aaccount_dict)
+        if aaccount_dict.get('recordset'):
+            return aaccount_dict['recordset']
+        if aaccount_dict.get('id'):
+            return aaao.browse(aaccount_dict['id'])
+        if aaccount_dict.get('code'):
+            aacode = aaccount_dict['code'].upper()
+            if aacode in speed_dict:
+                return aaao.browse(speed_dict[aacode])
+        raise self.user_error_wrap(_(
+            "Odoo couldn't find any analytic account corresponding to the "
+            "following information extracted from the business document: "
+            "Analytic account code: %s") % aaccount_dict.get('code'))
+
+    def _prepare_journal_speed_dict(self):
+        res = self.env['account.journal'].search_read([
+            ('company_id', '=', self.env.user.company_id.id)], ['code'])
+        speed_dict = {}
+        for l in res:
+            speed_dict[l['code'].upper()] = l['id']
+        return speed_dict
+
+    @api.model
+    def _match_journal(self, journal_dict, chatter_msg, speed_dict=None):
+        """Example:
+        journal_dict = {
+            'code': 'MISC',
+            }
+        speed_dict is usefull to gain performance when you have a lot of
+        journals to match
+        """
+        if not journal_dict:
+            journal_dict = {}
+        ajo = self.env['account.account']
+        if speed_dict is None:
+            speed_dict = self._prepare_journal_speed_dict()
+        self._strip_cleanup_dict(journal_dict)
+        if journal_dict.get('recordset'):
+            return journal_dict['recordset']
+        if journal_dict.get('id'):
+            return ajo.browse(journal_dict['id'])
+        if journal_dict.get('code'):
+            jcode = journal_dict['code'].upper()
+            if jcode in speed_dict:
+                return ajo.browse(speed_dict[jcode])
+            # case insensitive
+        raise self.user_error_wrap(_(
+            "Odoo couldn't find any journal corresponding to the "
+            "following information extracted from the business document: "
+            "Journal code: %s") % journal_dict.get('code'))
+
+    def get_xml_files_from_pdf(self, pdf_file):
+        """Returns a dict with key = filename, value = XML file obj"""
+        logger.info('Trying to find an embedded XML file inside PDF')
+        res = {}
+        try:
+            fd = StringIO(pdf_file)
```

Comment:
I think you need `BytesIO` here. I get an error otherwise, could you also check?

Reply:
@astirpe can you past error ? import test file are working ...

Reply:
I'm not able to reproduce it right now, I will be back to you in case it occurs again. Sorry.

Reply:
Sorry, I remember now...
There is a `try` that prevents you to see any error.
So just place your debugger breakpoint on that line and notice that it always fails: it always enters the `except` block, which simply says `pass`. That's why you don't see any error.

`BytesIO` seems fixing that behavior.

Reply:
Ok thanks. it's done